var/spack/repos/builtin/packages/scorep/package.py

```diff
@@ -72,6 +74,7 @@ def install(self, spec, prefix):
                 "--with-papi-header=%s" % spec['papi'].prefix.include,
                 "--with-papi-lib=%s" % spec['papi'].prefix.lib,
                 "--enable-shared",
+                "--without-shmem" if '~shmem' in spec else '',
```

Comment:
Oh how convenient.  It's already in #2386.  And it's called `with_or_without`.  @alalazo told me he is back to finishing that one so maybe after that is merged we can refactor.